planner/core/optimizer.go

```diff
@@ -396,6 +435,14 @@ func logicalOptimize(ctx context.Context, flag uint64, logic LogicalPlan) (Logic
 			continue
 		}
 		opt.appendBeforeRuleOptimize(i, rule.name(), logic)
+		// sync-load full stats before the first stats-needed rule applied
+		if !fullStatsLoaded && rule.needStats() {
+			_, err = SyncLoadColumnFullStats(logic)
+			if err != nil {
+				return nil, err
+			}
+			fullStatsLoaded = true
```

Comment:
If no rule needs stats for some plan, we need also load column stats after logical optimization?

Reply:
Good catch! Thx.

Reply:
Refactored and removed.